runtime/compiler/ilgen/Walker.cpp

```diff
@@ -4132,6 +4108,33 @@ TR_J9ByteCodeIlGenerator::genInvokeHandle(TR::SymbolReference *invokeExactSymRef
          TR_VerboseLog::writeLineLocked(TR_Vlog_MHD, "Call to invokeExact%.*s from %s", callee->signatureLength(), callee->signatureChars(), comp()->signature());
          }
       }
+
+   if (!_methodHandleInvokeNodes)
+      _methodHandleInvokeNodes = new (trHeapMemory()) TR::NodeChecklist(comp());
+
+   _methodHandleInvokeNodes->add(callNode);
+
+   return callNode;
+   }
+
+TR::Node *
+TR_J9ByteCodeIlGenerator::genILGenMacroInvokeExact(TR::SymbolReference *invokeExactSymRef)
+   {
+   if (comp()->compileRelocatableCode())
+      {
+      comp()->failCompilation<J9::AOTHasInvokeHandle>("COMPILATION_AOT_HAS_INVOKEHANDLE 1");
+      }
+
+   if (comp()->getOption(TR_FullSpeedDebug) && !isPeekingMethod())
+      comp()->failCompilation<J9::FSDHasInvokeHandle>("FSD_HAS_INVOKEHANDLE 1");
```

Comment:
I think these could be assertions rather than conditional compilation failures. AOT or FSD compilations should have failed before we start trying to `runMacro()`/`runFEMacro()`